xarray/core/dataarray.py

```diff
@@ -1385,6 +1385,13 @@ def reindex(
         align
         """
         indexers = either_dict_or_kwargs(indexers, indexers_kwargs, "reindex")
+        if isinstance(fill_value, dict):
```

Comment:
Just to make sure I understand -- what is the use case for dict fill_value for DataArray? Handling coordinates?

Reply:
yes, non-dimension coordinates. Honestly, when I added that I didn't think too much about this; I simply wanted to make sure the methods on both classes stay consistent.